lib/SPIRV/libSPIRV/SPIRVEntry.cpp

```diff
@@ -388,11 +388,11 @@ SPIRVEntry::setLinkageType(SPIRVLinkageTypeKind LT) {
   addDecorate(new SPIRVDecorateLinkageAttr(this, Name, LT));
 }
 
-std::ostream &
-operator<<(std::ostream &O, const SPIRVEntry &E) {
+llvm::raw_ostream &
+operator<<(llvm::raw_ostream &O, const SPIRVEntry &E) {
   E.validate();
   E.encodeAll(O);
-  O << SPIRVNL;
+  SPIRVNL(O);

```

Comment:
I implemented that solution, but it doesn't work.
I need to dig into it more deeply, but it's clear that due to overloading rules this function is not called. llvm::raw_ostream has another candidate, which C++ compiler likes better:

raw_ostream &raw_ostream::operator<<(const void *P) {
  *this << '0' << 'x';

  return write_hex((uintptr_t) P);
}

as you can see - it doesn't work for SPIR-V conversion.